### ISSUE TITLE
AnimationBuilder query returns zero elements when ":enter" or ":leave" is used

### ISSUE THREAD

**Issue by MurhafSousli** [angular/angular#26612]
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior

I am using **AnimationBuilder** to apply animations on child elements'  `:enter` and `:leave` transitions, I tried queries like as the ones in [the query docs](https://angular.io/api/animations/query#usage-notes) 

Using `:enter .child-element`  and `.child-element:enter` throw the following error

> ERROR
Error: Unable to create the animation due to the following errors: `query(":enter .child-element")` returned zero elements. (Use `query(":enter .child-element", { optional: true })` if you wish to allow this.)

## Expected behavior

Ability to query child elements with `:enter` and `:leave` transitions

## Minimal reproduction of the problem with instructions

```ts
const builder = this.animationBuilder.build([
  query(
    '.child-element',          // Works
//  ':enter .child-element', // This does not work !
//  '.child-element:enter',  // This also does not work !
    [
     stagger(100, [useAnimation(animation)])
    ]
  )
]);
```

https://stackblitz.com/edit/angular-jh4gnd?file=src%2Fapp%2Fapp.component.ts

## Environment

<pre><code>
Angular version: 7.0.0

Browser:
- [x ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 8
- Platform:  Windows
Others:

</code></pre>

**Comment by dario-piotrowicz** [comment]
@MurhafSousli I do not think this is supposed to work, queries (or better yet transitions) with `:enter` and `:leave` are usually used with the `animation` field in the `@Component` decorator and with them you can ask Angular to do something reactively when the DOM changes (it notices when new elements enter the DOM, when old ones are removed and triggers your animations accordingly).

In your example instead are manually playing an animation, and the query gets executed in the precise moment in which you run the `play()` method, but in that moment the elements are all already in the DOM so I do not really think you should be able to query them.

Basically what I am saying is that I think that `:enter` and `:leave` are to be used in the decorator for reactive behaviors, as in, you declare them and let Angular react and handle them accordingly when there are changes in the DOM. But when you yourself programmatically ask an animation to run your query, that doesn't gets executed when Angular sees some DOM changes, but when you run the `play()` method instead, which regardless on timing is distinct and happens at a different time compared to the DOM rendering so when you run the method it is impossible (or almost) to get those elements queried since they are inserted/removed at a different time.

(In other words, I think that `:enter`/`:leave` are by design __declarative__ whilst the players build and played in this way are clearly __procedural__ so the two do not match well together)

**Comment by thePunderWoman** [comment]
This issue was closed as obsolete due to the introduction of the new `animate.enter` and `animate.leave` APIs. The `@angular/animations` package is now deprecated and will be removed in Angular v23. We hope you give the new APIs a try.
